### ISSUE TITLE
BUG: GenericBackend always uses tied embedding for lm_head — never loads output.weight

### ISSUE THREAD

**Issue by bong-water-water-bong** [1bit-MONSTER/1bit-MONSTER#319]
## Summary

`src/backend_generic.cpp`'s `forward()` computes final logits unconditionally as:

```cpp
// LM head (tied embedding)
matmul(logits_buf.data(), x2.data(), embed.data(), V, H);
```

This reuses `token_embd.weight` for the LM head in every case. `load_gguf()` never looks for a separate `output.weight` tensor at all — it isn't in the tensor-loading loop anywhere.

## Why this is a bug

Embedding tying (`tie_word_embeddings: true` in HF configs) is a per-model choice, not a given. When a model does **not** tie embeddings, `output.weight` holds a genuinely different matrix from `token_embd.weight`. For such models this code silently computes logits with the wrong weights — no error, no warning, just wrong (likely incoherent) generated text.

## Verification

Checked against a real Qwen3-0.6B GGUF (converted via `convert_hf_to_gguf.py`): `token_embd.weight` and `output.weight` happen to be byte-identical in that specific checkpoint, so this bug is currently masked for that model. But that's incidental to this one checkpoint's tie setting, not something the code checks for — any untied model (common for larger-vocab or larger models) hits the bug.

## Fix

In `load_gguf()`, attempt to load `output.weight` (shape `[hidden, vocab]`, same as `token_embd.weight`) as a separate optional tensor. If present, use it for the LM head matmul instead of `embed.data()`; if absent, fall back to tied embedding (current behavior) — mirroring the `load_tensor_optional` pattern already used for QKV bias and Q/K-norm in the same file.

## Context

Found while verifying #15's Q/K-norm + MoE work (numerical reference-check against a real dense Qwen3-0.6B GGUF) — out of scope for that change, filing separately per repo convention.

## Severity

**Medium.** Silent wrong output on any untied-embedding model routed to `cpu_generic`, no error surfaced.
